src/ui/SmartSnippet/SmartSnippet.ts

```diff
@@ -301,19 +301,21 @@ export class SmartSnippet extends Component {
   }
 
   private renderSourceTitle() {
-    return this.buildLink(Utils.getFieldValue(this.lastRenderedResult, <string>this.options.titleField), SOURCE_TITLE_CLASSNAME);
+    const link = this.buildLink();
+    link.innerText = Utils.getFieldValue(this.lastRenderedResult, <string>this.options.titleField);
+    link.classList.add(SOURCE_TITLE_CLASSNAME);
+    return link;
   }
 
   private renderSourceUrl() {
-    const uri = this.options.hrefTemplate
-      ? StringUtils.buildStringTemplateFromResult(this.options.hrefTemplate, this.lastRenderedResult)
-      : this.lastRenderedResult.clickUri;
```

Comment:
The removed logic was essentially just emulating what the result link does. It should no longer be needed.